prometheus/handler.go

```diff
@@ -45,8 +46,21 @@ type Handler struct {
 	// If nil, stats.Buckets is used instead.
 	Buckets stats.HistogramBuckets
 
-	opcount uint64
-	metrics metricStore
+	opcount       uint64
+	metrics       metricStore
+	ignoredLabels []byte
+}
+
+// IgnoreLabels will remove the passed labels from the observation
+func (h *Handler) IgnoreLabels(labelNames []string) {
+	b := make([][]byte, len(labelNames))
+	for i := range labelNames {
+		b[i] = []byte(labelNames[i])
+	}
+	// This flattens all the ignored labels into a single byte slice
+	// so we dont have to do an explicit O(n^2) lookup when we filter
+	// labels out of the observation
+	h.ignoredLabels = bytes.Join(b, []byte{0x00})
```

Comment:
Hmm, this seems unnecessary.  Whether or not you're iterating over a slice of strings or doing `bytes.Contains` shouldn't really have a significant performance difference, and the former is much more clear.  They're both `O(n*m)` where n is the number of tags and m is the number of ignored, not `O(n**2)`.

If we add this, I would suggest using a list of strings representing the labels instead.

Reply:
Thanks @abraithwaite.  Your suggested changes have been submitted